src/cpp/r/R/Api.R

```diff
@@ -841,6 +841,33 @@ options(terminal.manager = list(terminalActivate = .rs.api.terminalActivate,
    .Call("rs_systemUsername", PACKAGE = "(embedding)")
 })
 
+# store callback functions to be executed after a specified chunk
+# and return a handle to unregister the chunk
+.rs.addApiFunction("registerChunkCallback", function(chunkCallback = function(chunkName, chunkCode){}) {
+
+   if (!is.function(chunkCallback))
+      stop("'chunkCallback' must be a function")
+
+   # if one does not already exist, add an environment which will host registered callbacks
+   if (!exists(".rs.notebookChunkCallbacks"))
```

Comment:
For robustness you should check if the environment `exists` specifically in the tools environment.